server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -477,13 +477,12 @@ public GetUploadParamsResponse doInTransaction(TransactionStatus status) throws
                 //using the existing max upload size configuration
                 command.setProcessTimeout(NumbersUtil.parseLong(_configDao.getValue("vmware.package.ova.timeout"), 3600));
                 command.setMaxUploadSize(_configDao.getValue(Config.MaxUploadVolumeSize.key()));
-                command.setAccountId(vol.getAccountId());
-                Account account = _accountDao.findById(vol.getAccountId());
-                if (account.getType().equals(Account.Type.PROJECT)) {
-                    command.setDefaultMaxSecondaryStorageInGB(ResourceLimitService.MaxProjectSecondaryStorage.value());
-                } else {
-                    command.setDefaultMaxSecondaryStorageInGB(ResourceLimitService.MaxAccountSecondaryStorage.value());
-                }
+
+                Long accountId = vol.getAccountId();
+                Account account = _accountDao.findById(accountId);
+
+                command.setDefaultMaxSecondaryStorageInGB(_resourceLimitMgr.findCorrectResourceLimitForAccount(account, ResourceType.secondary_storage));
```

Comment:
@JoaoJandre 
I think the domain resource limitation should be considered as well, as the account resource limitation might exceed the domain resource limitation.

Reply:
@weizhouapache I did some testing and confirmed this, also tested to see if this kind of check is done for other things, like number of vm instances, and the domain limit is considered in that case. So I added the domain resource limitation check to the secondary storage as well.